examples/ShowLoader.cpp

```diff
@@ -141,6 +141,24 @@ ShowLoader::State ShowLoader::NextFrame(unsigned int *universe,
 }
 
 
+/**
+ * Read the next show file entry
+ * @param entry a ShowEntry to fill with data
+ */
+ShowLoader::State ShowLoader::NextEntry(ShowEntry *entry) {
+  State state = NextFrame(&entry->universe, &entry->buffer);
+  if (state != State::OK) {
+      return state;
+  }
+  state = NextTimeout(&entry->next_wait);
```

Comment:
Technically, if a show file only has one entry, there will be no timeout, just:
`1 1,2,3`

Reply:
This may actually be a bug in the playback post-refactor - since the last entry never has a timeout, the last frame will never be sent.  I'll have to rework this a bit to accommodate this.

Reply:
Would you say it's safe to assume a show file with a trailing timeout is invalid?  Assuming a showfile will always end with a frame makes this less messy.  It looks like verify didn't care before, but the recorder won't output a trailing timeout.

Reply:
There's a chance it might have existed before too, I've not reviewed the code too closely.

I added this TODO note in my other PR:
https://github.com/OpenLightingProject/ola/pull/1607/files#diff-5970e4574d66fe51e161810176e44cadR110-R111

I think there are lots of edge cases, but given we're reliant on incoming data that's not under our control to get the frames, I think the best case would be if it could handle leading and trailing timeouts too. Especially as the format is so simple, I've certainly suggested generating show files, I'm not sure I've actually done it. If you're recording a duration, do you want 10s from now, or from the first DMX frame or what? We could also theoretically grab the current data when we start recording, but that potentially changes the frequency of the frames giving a lumpy fade.

If you're doing the duration stuff, it's conceivable you want say a 10s, or an hour long recording, and you don't want that to drift over a week, so the fancy display at the top of the hour is always at that time.

I think at a minimum verify should throw a warning/error and report statistics that match what it will actually do in terms of timing. We could always leave fixing the complicated bit for the future with a TODO or an issue.

Reply:
Alright, it's a bit clunky, but I think I've found a way to handle both cases (trailing timeout/trailing frame).

I generated a few test show file by having sACNView run a chase across two universes; having about 10 seconds of data to play with made testing easier.

I might make a separate PR that cleans up output - as I've been testing this I've used my desk calculator to add up the timeouts to verify that start/stop times are what they should be.  For DMX output, I feel something along the lines of "<time>: Univ <#>: <dmx values> would be more helpful, but that also feels a bit out of scope here.  As you said, if you use this for something long-running it's conceivable the clock might drift quite a bit (especially on something nasty without an actual RTC like a Raspberry Pi).